### PR TITLE
datastore: Prevent unicode/ascii conversion errors

### PR DESCRIPTION
**Pull request by mordae**
Some error messages were causing the errors due to format conversion of unicode fields into str templates.

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [X] includes bugfix for possible backport

### CODE REVIEW

File: ckanext/datastore/db.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def _cache_types(context):
         if 'nested' not in _type_names:
             native_json = _pg_version_is_at_least(connection, '9.2')
 
-            log.info("Create nested type. Native JSON: {0}".format(
+            log.info(u"Create nested type. Native JSON: {0}".format(
```
Comment (wardi) [inline ckanext/datastore/db.py]: How about using repr() for user-supplied values sent to logs. That seems safer.

**Comment by mordae** [comment]
You are right, it's usually not a good idea.

**Comment by TkTech** [comment]
@mordae `{0!r}` is the same as `%r/repr(x)`

**Comment by mordae** [comment]
OK, let's make it simpler. These are just the `ValidationError` formats.

File: ckanext/datastore/db.py [hunk 2]
```diff
@@ -1310,8 +1313,7 @@ def make_private(context, data_dict):
 
 
 def make_public(context, data_dict):
-    log.info('Making resource {0} public'.format(
-        data_dict['resource_id']))
+    log.info('Making resource {resource_id!r} public'.format(**data_dict))
```
Comment (wardi) [inline ckanext/datastore/db.py]: have you been using something other than UUIDs for your resource_ids?

Comment (mordae) [inline ckanext/datastore/db.py]: Nope, just being consistent and quoting all of them. It's `UnicodeText` in the model...

**Comment by wardi** [comment]
@amercader This puts python reprs in our logs now, do you think that's a concern for backporting?

**Comment by mordae** [comment]
As for the `ValidationError` fixes; yes, the unicode-related crashes are easily reproducible. As for the other commit I frankly do not know. Thanks for the merge, by the way. :-)

**Pull request merged**